addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -320,32 +320,6 @@ static func get_general_blocks() -> Array[Block]:
 
 #endregion
 #region Variables
-
-	b = BLOCKS["statement_block"].instantiate()
-	b.block_format = "Set String {var: STRING} {value: STRING}"
-	b.statement = "VAR_DICT[{var}] = {value}"
```

Comment:
Will removing support for `VAR_DICT` break existing serializations? Do we need some kind of migration routine to handle it?

Reply:
The way we are serialize blocks makes it so that (unfortunately) we can't modify blocks once they are placed in the scene. [T35536](https://phabricator.endlessm.com/T35536) will hopefully fix this.

We could have a migration routine, but I think it's probably best we just spend time finishing that ticket so we don't have to worry about making similar routines in the future.

This PR does break the sound blocks though... should I append a commit here to fix them? It's a rather large change so I could make another PR and just disable them here temporarily.

Reply:
I think the sound blocks is kinda the point, right? If it's happening in our blocks it's certainly going to happen for others. What happens when you have a scene with a block script and a variable of some kind and then you upgrade the plugin?

I'd like to see the commit to see what's going to be required. It's early enough where we could say "sorry, your script might be broken", but at least I think we need to know what you'd have to do to fix it.

Reply:
Yeah, when you upgrade the plugin, the blocks in each script don't get upgraded, you have to manually go in and replace each of them if you want the new ones from the picker... which is not ideal. So if anyone has used a `VAR_DICT` style variable in their scenes, it will break.

This has happened a few times in the past when we have upgraded blocks, and even the pong game example is pretty outdated, though the old blocks still work. 

I suppose we could throw in something to the code generator that detects if a `VAR_DICT` statement is in the script and adds it back to the top for legacy scripts. I guess I just don't want the old `VAR_DICT` stuff to haunt me in the current code, but I can see why it's good to have support for older versions 😄

Reply:
I don't think you'd try to keep the `VAR_DICT` statements. I think you'd detect them and rewrite them using the new syntax. I don't know how difficult that would be, but it's something I think we're going to end up dealing with more than once.

The serialization stored in the project's resource files are essentially ABI. If we break it, we're going to annoy our users. So, we need to get it in the shape we want ASAP, but we also need to be honest that we're going to find issues with it (like we are now) and that we have to have a plan for managing those transitions.

There will always be problems that come up that affect the serialization, whether we want it to or not. I'm not saying you personally need to come up with the plan to handle it. It's something the whole team needs to think about and be aware of as we continue iterating. To say it another way, the serialization is maybe the most critical part of the whole plugin because changes in it immediately break users' projects. Nothing could be more annoying than updating our plugin and finding out your project is now broken.

Reply:
> I don't think you'd try to keep the VAR_DICT statements. I think you'd detect them and rewrite them using the new syntax. I don't know how difficult that would be, but it's something I think we're going to end up dealing with more than once.

Okay, I created https://phabricator.endlessm.com/T35547 about adding a migration system. Indeed, it's a good thing to do early before we really need it :) I don't think we need to block on adding migrations for this particular case this early in the thing's life, but then it _could_ be a useful test case.

Reply:
After talking to Daniel, he suggested it was too early to worry about backwards compatibility since we haven't even had this in front of learners yet. So, we can just ignore this for now, but it's definitely something to keep in mind going forward.